### DISCUSSION TITLE
AppVersionRelease required even if specified in argument

### DISCUSSION THREAD

**Discussion by mysteryx93** [kuiperzone/PupNet-Deploy#20] [Q&A]
App version can be specified with `-v` argument; but even there, it complains that `AppVersionRelease` is required.

If both are specified, is `AppVersionRelease` ignored to be replaced by the specified version?

I don't want to have to modify the `.conf` file for every release.

**Comment by kuiperzone** [comment]
>App version can be specified with -v argument; but even there, it complains that AppVersionRelease is required.

AppVersionRelease is a mandatory field in the conf file, and is checked separately to the command line. However, if you wish to provide it only on the command line, simply set AppVersionRelease to "1.0.0" and forget about it.

>If both are specified, is AppVersionRelease ignored to be replaced by the specified version?

Yes.

As per conf documentation:

Mandatory application version and package release of form: 'VERSION[RELEASE]'. Use optional square
brackets to denote package release, i.e. '1.2.3[1]'. Release refers to a change to the deployment
package, rather the application. If release part is absent (i.e. '1.2.3'), the release value defaults
to '1'. Note that the version-release value given here may be overridden from the command line.
Example: AppVersionRelease = 1.0.0[1]
